### DISCUSSION TITLE
How to connect to multiple clusters in the same client?

### DISCUSSION THREAD

**Discussion by eb2tech** [dotnet/orleans#8478] [Q&A] [answered]
Prior to v7.x, we were able to use a ClientBuilder to concurrently connect to multiple clusters. How is this best achieved now that ClientBuilder has been removed in v7.x?

Thanks,
Ed

**Comment by ReubenBond** [comment, marked as answer]
It's similar in 7.x. Think of it as the ClientBuilder being unbundled from the HostBuilder.

For each client, you would create a new HostBuilder and then add the client to it

**Reply by eb2tech** [reply]
Let me see if I understand. Our 3.x implementation essentially had a dictionary of IClusterClients, keyed by cluster name. If a request for a cluster was made and an IClusterClient didn't already exist, one would be created using a ClientBuilder. With 7.x, we should create a HostBuilder instead, AddOrleansClient to it, and request the configured IClusterClient from the set of services built for that Host, adding that IClusterClient to the dictionary?

**Reply by eb2tech** [reply]
Or start with the existing IServiceCollection? (I admit I'm still learning how to use the new host extension mechanisms. The company I work for is just making the move to net6.0+.) I think I have something to start with.

**Reply by eb2tech** [reply]
One other caveat is that a silo in, say, cluster A, needs to act as a client to cluster B. A sample demonstrating how this would work under 7.x would be **greatly** appreciated.
